ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSmain.cpp

```diff
@@ -463,12 +463,17 @@ void execute_program(std::string fname, std::string args, bool wait) { execute_p
 
 // converts a relative path to absolute if the path exists
 std::string filename_absolute(std::string fname) {
-  if (file_exists(fname) || directory_exists(fname)) {
-    wchar_t rpath[MAX_PATH];
-    tstring tstr_fname = widen(fname);
-    GetFullPathNameW(tstr_fname.c_str(), MAX_PATH, rpath, NULL); 
-    return shorten(rpath);
-  } else { return ""; }
+  if (string_replace_all(fname, " ", "") == "") fname = ".";
+  wchar_t rpath[MAX_PATH];
+  tstring tstr_fname = widen(fname);
+  GetFullPathNameW(tstr_fname.c_str(), MAX_PATH, rpath, NULL);
```

Comment:
If this method fails, it will return zero. Regardless, it returns the length of the result string. It would be good to use that length to construct a tstring, ie,
```cpp
tstring result(rpath, GetFullPathNameW(tstr_fname.c_str(), MAX_PATH, rpath, NULL));
return shorten(result);
```

Since the system method already has to check disk to evaluate this path, we can assume the file exists when it completes, so checking if `file_exists` doesn't help us.

Also, why do we care if the string ends in a slash? You can refer to a directory with or without the slash.

Reply:
I want all functions that return existing paths to include the trailing slash in case the user doesn't specify it. for example, our working_directory includes it:

`example = working directory + "file.png"`

If we didn't add the trailing slash, this code wouldn't work, and would also be incompatible with how GMS 1.4 and 2.x does things.